ExtractUserDataActivity/handler.ts

```diff
@@ -194,63 +181,64 @@ const logFailure = (context: Context) => (
   profileModel: ProfileModel,
   fiscalCode: FiscalCode
-): TaskEither<
+): TE.TaskEither<
   ActivityResultUserNotFound | ActivityResultQueryFailure,
   Profile
 > =>
-  profileModel
-    .findLastVersionByModelId([fiscalCode])
-    .foldTaskEither<
-      ActivityResultUserNotFound | ActivityResultQueryFailure,
-      Profile
-    >(
-      failure =>
-        fromLeft(
-          ActivityResultQueryFailure.encode({
-            kind: "QUERY_FAILURE",
-            reason: `${failure.kind}, ${getMessageFromCosmosErrors(failure)}`
+  pipe(
+    profileModel.findLastVersionByModelId([fiscalCode]),
+    TE.mapLeft(failure =>
+      ActivityResultQueryFailure.encode({
+        kind: "QUERY_FAILURE",
+        reason: `${failure.kind}, ${getMessageFromCosmosErrors(failure)}`
+      })
+    ),
+    TE.chainW(
+      flow(
+        E.fromOption(() =>
+          ActivityResultUserNotFound.encode({
+            kind: "USER_NOT_FOUND_FAILURE"
           })
         ),
-      maybeProfile =>
-        fromEither<ActivityResultUserNotFound, Profile>(
-          fromOption(
-            ActivityResultUserNotFound.encode({
-              kind: "USER_NOT_FOUND_FAILURE"
-            })
-          )(maybeProfile)
-        )
-    );
+        TE.fromEither
+      )
+    )
+  );
 /**
  * Retrieves all contents for provided messages
  */
 export const getAllMessageContents = (
   messageContentBlobService: BlobService,
   messageModel: MessageModel,
   messages: ReadonlyArray<RetrievedMessageWithoutContent>
-): TaskEither<
+): TE.TaskEither<
   ActivityResultQueryFailure,
   ReadonlyArray<MessageContentWithId>
 > =>
-  array.sequence(taskEither)(
-    messages.map(({ id: messageId }) =>
-      messageModel
-        .getContentFromBlob(messageContentBlobService, messageId)
-        .foldTaskEither<ActivityResultQueryFailure, MessageContentWithId>(
-          _ => fromEither(right({ messageId } as MessageContentWithId)),
-          maybeContent =>
-            fromEither(
-              maybeContent.foldL(
-                () => right({ messageId } as MessageContentWithId),
-                (content: MessageContent) =>
-                  right({
-                    content,
-                    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
-                    messageId: messageId as NonEmptyString
-                  })
-              )
-            )
+  pipe(
+    messages,
+    A.map(_ => _.id),
+    A.map(messageId =>
+      pipe(
+        messageModel.getContentFromBlob(messageContentBlobService, messageId),
+        TE.chainW(
+          flow(
+            E.fromOption(() => void 0 /* anything will do */),
+            E.map(content => ({
+              content,
+              messageId
+            })),
+            TE.fromEither
+          )
+        ),
+        TE.fold(
+          // in case of failure retrieving the single message, just live a placeholder
+          () => TE.of({ messageId } as MessageContentWithId),
```

Comment:
Not sure we need the explicit cast :)

Reply:
we do as `{ messageId }` ain't a valid `MessageContentWithId`. Which is acceptable in this case, as we are not processing such data later